source/features/restore-file.tsx

```diff
@@ -7,21 +8,45 @@ import * as pageDetect from 'github-url-detection';
 import features from '../feature-manager';
 import * as api from '../github-helpers/api';
 import showToast from '../github-helpers/toast';
-import {getConversationNumber} from '../github-helpers';
 import {getBranches} from '../github-helpers/pr-branches';
+import getPrInfo from '../github-helpers/get-pr-info';
+import {getConversationNumber} from '../github-helpers';
 
-// Get the current base commit of this PR. It should change after rebases and merges in this PR.
-// This value is not consistently available on the page (appears in `/files` but not when only 1 commit is selected)
 const getBaseReference = onetime(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository() {
-			pullRequest(number: ${getConversationNumber()!}) {
-				baseRefOid
-			}
-		}
-	`);
-	return repository.pullRequest.baseRefOid;
+	const {base} = getBranches();
+	const {baseRefOid} = await getPrInfo(base.relative);
+	return baseRefOid;
 });
```

Comment:
I didn't used `onetime` on `headRefOid` of the Pull Request since `headRefOid` keeps changing when we restoring files continuously.